src/Orleans.Core/Streams/PubSub/ImplicitStreamSubscriberTable.cs

```diff
@@ -79,14 +82,15 @@ private Cache BuildCache(MajorMinorVersion version, ImmutableDictionary<GrainTyp
                             + $" Ensure that a corresponding {nameof(IStreamNamespacePredicateProvider)} is registered");
                     }
 
-                    bool includeNamespaceInGrainId = false;
-                    if (grainBinding.TryGetValue(WellKnownGrainTypeProperties.StreamBindingIncludeNamespaceKey, out var value)
-                        && string.Equals(value, "true", StringComparison.OrdinalIgnoreCase))
+                    if (!grainBinding.TryGetValue(WellKnownGrainTypeProperties.StreamIdMapperKey, out var mapperName))
                     {
-                        includeNamespaceInGrainId = true;
+                        throw new KeyNotFoundException(
+                           $"Stream binding for grain type {binding.GrainType} is missing a \"{WellKnownGrainTypeProperties.StreamIdMapperKey}\" value");
                     }
+                    var streamIdMapper = _serviceProvider.GetServiceByName<IStreamIdMapper>(string.IsNullOrWhiteSpace(mapperName) ? DefaultStreamIdMapper.Name : mapperName);
```

Comment:
I think we should pass the default name through the grain bindings, rather than pass null/empty and map it here